Makefile.uk

```diff
@@ -0,0 +1,199 @@
+LIBSODIUM_LIBRARY_VERSION_MAJOR=10
+LIBSODIUM_LIBRARY_VERSION_MINOR=3
+LIBSODIUM_URL=https://github.com/jedisct1/libsodium/archive/refs/tags/$(LIBSODIUM_VERSION).zip
+LIBSODIUM_SUBDIR=libsodium-$(LIBSODIUM-VERSION)
+LIBSODIUM_PATCHDIR=$(LIBSODIUM_BASE)/patches
+$(eval $(call fetch,libsodium,$(LIBSODIUM_URL)))
+$(eval $(call patch,libsodium,$(LIBSODIUM_PATCHDIR),$(LIBSODIUM_SUBDIR)))
+
+################################################################################
+# Helpers
+################################################################################
+LIBSODIUM_EXTRACTED = $(LIBSODIUM_ORIGIN)/libsodium-$(LIBSODIUM_VERSION)
+
+################################################################################
+# Library includes
+################################################################################
+LIBSODIUM_COMMON_INCLUDES-y         += -I$(LIBSODIUM_EXTRACTED)/src/libsodium/include/sodium
+LIBSODIUM_CINCLUDES                 += $(LIBSODIUM_COMMON_INCLUDES-y)
+LIBSODIUM_CXXINCLUDES               += $(LIBSODIUM_COMMON_INCLUDES-y)
+
+CINCLUDES-$(CONFIG_LIBSODIUM)   += -I$(LIBSODIUM_EXTRACTED)/src/libsodium/include
+CXXINCLUDES-$(CONFIG_LIBSODIUM) += -I$(LIBSODIUM_EXTRACTED)/src/libsodium/include
+
+################################################################################
+# Global flags
+################################################################################
+LIBSODIUM_SUPPRESS_FLAGS += -fno-strict-aliasing -fno-strict-overflow
+
+LIBSODIUM_CFLAGS-y      += $(LIBSODIUM_SUPPRESS_FLAGS)
+LIBSODIUM_CXXFLAGS-y    += $(LIBSODIUM_SUPPRESS_FLAGS)
+
+LIBCRYPTO_CFLAGS-y   += $(LIBSODIUM_SUPPRESS_FLAGS)
+LIBCRYPTO_CXXFLAGS-y += $(LIBSODIUM_SUPPRESS_FLAGS)
+
+# Extracted from original build by running configure --minimal and removing the following:
+#
+# -DHAVE_SYS_RANDOM_H=1 -DHAVE_ALLOCA_H=1 -DHAVE_ALLOCA=1 -DHAVE_MMAP=1 -DHAVE_MLOCK=1 -DHAVE_MADVISE=1 -DHAVE_MPROTECT=1
+# -DHAVE_GETRANDOM=1 -DHAVE_GETENTROPY=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_GETPID=1 -DHAVE_NANOSLEEP=1 -DHAVE_EXPLICIT_BZERO=1 
+#
+LIBSODIUM_DEFINES += -DPACKAGE_NAME=\"libsodium\" -DPACKAGE_TARNAME=\"libsodium\" -DPACKAGE_VERSION=\"1.0.18\" -DPACKAGE_STRING=\"libsodium\ 1.0.18\" \
+ -DPACKAGE_BUGREPORT=\"https://github.com/jedisct1/libsodium/issues\" -DPACKAGE_URL=\"https://github.com/jedisct1/libsodium\" -DPACKAGE=\"libsodium\" \
+ -DVERSION=\"1.0.18\" -DMINIMAL=1 -DHAVE_PTHREAD_PRIO_INHERIT=1 -DHAVE_PTHREAD=1 -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 \
+ -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -D__EXTENSIONS__=1 -D_ALL_SOURCE=1 \
+ -D_GNU_SOURCE=1 -D_POSIX_PTHREAD_SEMANTICS=1 -D_TANDEM_SOURCE=1 -DHAVE_C_VARARRAYS=1 -DTLS=_Thread_local -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" \
+ -DHAVE_SYS_MMAN_H=1 -DNATIVE_LITTLE_ENDIAN=1 -DHAVE_INLINE_ASM=1 -DHAVE_TI_MODE=1 -DASM_HIDE_SYMBOL=.hidden -DHAVE_WEAK_SYMBOLS=1 -DHAVE_ATOMIC_OPS=1  -DCONFIGURED=1
```

Comment:
Can you break this up into multiple lines per definition? I think it will make it easier to read :-) E.g.:
```Makefile
LIBSODIUM_DEFINES += -DPACKAGE_NAME=\"libsodium\" \
                     -DPACKAGE_TARNAME=\"libsodium\" \
                     ...
```

Additionally, I notice that the version number is inline here, please can you add it dynamically, i.e.:

```diff
                     ...
-                    -DPACKAGE_VERSION=\"1.0.18\" \
+                    -DPACKAGE_VERSION=\"$(LIBSODIUM_VERSION)\" \
```

Reply:
Done. Also made the dependency to pthread-embedded optional.